pageserver/src/tenant.rs

```diff
@@ -2933,20 +2977,12 @@ impl Tenant {
                         .0,
                 );
 
-                match gc_cutoffs.remove(&timeline.timeline_id) {
-                    Some(cutoffs) => {
-                        target.retain_lsns = branchpoints;
-                        target.cutoffs = cutoffs;
-                    }
-                    None => {
-                        // reasons for this being unavailable:
-                        // - this timeline was created while we were finding cutoffs
-                        // - lsn for timestamp search fails for this timeline repeatedly
-                        //
-                        // in both cases, refreshing the branchpoints is correct.
-                        target.retain_lsns = branchpoints;
-                    }
-                };
+                // Apply the cutoffs we found to the Timeline's GcInfo.  Why might we _not_ have cutoffs for a timeline?
+                // - this timeline was created while we were finding cutoffs
+                // - lsn for timestamp search fails for this timeline repeatedly
+                if let Some(cutoffs) = gc_cutoffs.remove(&timeline.timeline_id) {
```

Comment:
Pre-existing, but still: why can we `.remove()` here and be certain that this timeline_id won't be `.get()`ed above in

https://github.com/neondatabase/neon/blob/101fb6544d87e9f5442727ee20e06f5bed969dd4/pageserver/src/tenant.rs#L2955-L2961

Reply:
I think you're right... it wasn't noticeable because the .get() above was just an advisory metric.

Changed this to a .get() in 6d450f03ed1f5232a636a86d4414db799a70f438